hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -156,11 +159,56 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
     /**
      * Validate the fields in the transaction body that involves checking with dynamic
      * properties or state. This check is done as part of the handle workflow.
-     * @return OK if the transaction body is valid, otherwise return the appropriate error code
+     * @param context handle context
+     * @param accountStore account store
+     * @param op crypto create transaction body
      */
-    private ResponseCodeEnum validateSemantics() {
-        // FUTURE : Need to add validations that involve dynamic properties or state
-        return OK;
+    private void validateSemantics(
+            @NonNull final HandleContext context,
+            @NonNull final WritableAccountStore accountStore,
+            @NonNull final CryptoCreateTransactionBody op) {
+
+        final var cryptoCreateWithAliasConfig =
+                context.configuration().getConfigData(CryptoCreateWithAliasConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var entitiesConfig = context.configuration().getConfigData(EntitiesConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+
+        // validate payer account exists and has enough balance
+        final var payer = accountStore.getForModify(
+                context.body().transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT));
+
+        validateTrue(payer != null, INVALID_PAYER_ACCOUNT_ID);
+        final long newPayerBalance = payer.tinybarBalance() - op.initialBalance();
+        validatePayer(payer, newPayerBalance);
+
+        context.attributeValidator().validateMemo(op.memo());
+        cryptoCreateValidator.validateKeyAliasAndEvmAddressCombinations(
+                op, context.attributeValidator(), cryptoCreateWithAliasConfig, accountStore);
+        context.attributeValidator()
+                .validateAutoRenewPeriod(op.autoRenewPeriod().seconds());
+        validateFalse(
+                cryptoCreateValidator.tooManyAutoAssociations(
+                        op.maxAutomaticTokenAssociations(), ledgerConfig, entitiesConfig, tokensConfig),
+                REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
+        validateFalse(
+                op.hasProxyAccountID() && !op.proxyAccountID().equals(AccountID.DEFAULT),
+                PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED);
+
+        stakingValidator.validateStakedId(
+                op.declineReward(),
+                op.stakedId().kind().name(),
+                op.stakedAccountId(),
+                op.stakedNodeId(),
+                accountStore,
+                context,
+                nodeInfo);
+
+        // Change payer's balance to reflect the deduction of the initial balance for the new
+        // account
+        final var modifiedPayer =
+                payer.copyBuilder().tinybarBalance(newPayerBalance).build();
+        accountStore.put(modifiedPayer);
```

Comment:
`validateSemantics()` is meant to check that a transaction can be handled, but due to the method name, it shouldn't change any state – it should only check the validity. Because it's in a place that doesn't match the method name, this code will be difficult to find.

Reply:
Moved changing state out of the method.